effekt/shared/src/main/scala/effekt/core/Tree.scala

```diff
@@ -125,8 +126,22 @@ case class Property(id: Id, tpe: BlockType) extends Tree
  * FFI external definitions
  */
 enum Extern extends Tree {
-  case Def(id: Id, tparams: List[Id], cparams: List[Id], vparams: List[Param.ValueParam], bparams: List[Param.BlockParam], ret: ValueType, annotatedCapture: Captures, body: Template[Pure])
-  case Include(contents: String)
+  case Def(id: Id, tparams: List[Id], cparams: List[Id], vparams: List[Param.ValueParam], bparams: List[Param.BlockParam], ret: ValueType, annotatedCapture: Captures, bodies: List[ExternBody])
+  case Include(featureFlag: FeatureFlag, contents: String)
+}
+sealed trait ExternBody extends Tree {
+  def featureFlag: FeatureFlag
+}
+object ExternBody {
+  case class StringExternBody(featureFlag: FeatureFlag, contents: Template[Pure]) extends ExternBody
+  case class EffektExternBody(featureFlag: FeatureFlag, body: Stmt) extends ExternBody
```

Comment:
Potentially risking conflicts with `String` this could be shortened to
```scala
enum ExternBody extends Tree {
  def featureFlag: FeatureFlag
  
  case String(featureFlag: FeatureFlag, contents: Template[Pure])
  case EffektExternBody(featureFlag: FeatureFlag, body: Stmt)
}
```

Reply:
I'm not sure it's worth it given the potential conflicts. In particular, adding any helper methods on `ExternBody` that use `String`s as parameters/return values would become awkward (and might not be that unlikely).
Maybe we can also come up with a better name (I couldn't, hence the not entirely helpful one... -s, actually).